### DISCUSSION TITLE
Can we open an existing word document and read the bookmarks / XML mapping fields

### DISCUSSION THREAD

**Discussion by AmeyDeshTQ** [dolanmiu/docx#2110] [Q&A] [answered]
Hi there,

We would like to read existing Word documents /templates which contain bookmarks or XML mapping fields  - see the below screenshot and replace those with dynamic values.  We also like to add a custom XML part to the word document

Does Docx.js support this feature? 

![image](https://user-images.githubusercontent.com/126790075/236453992-55ce4c4c-007a-43a7-9ffe-5a599e61ab8e.png)

Thanks for the work so far and best luck 

Amey

**Comment by dolanmiu** [comment, marked as answer]
Not possible im afraid
At the moment, can open to modify documents

**Reply by AmeyDeshTQ** [reply]
@dolanmiu  -Thanks for the prompt reply, Do you mean we can open the documents to update and not it's possible to update bookmarks and the XML  mapping fields? 

Are these features on your roadmap?

**Reply by dolanmiu** [reply]
Correct:

Check out the demo: https://github.com/dolanmiu/docx/blob/master/demo/85-template-document.ts

**Reply by hshoja** [reply]
@dolanmiu 
What about just reading the xml mapping? 
can document patching and XML styles be used simultaneously?

**Reply by dolanmiu** [reply]
no there isn't a way to read the xml mapping (I think you mean xml file?) You can simply un-zip it and read it yourself. No need for a library
